### PR TITLE
[tests-only][full-ci] adding test coverage for backup consistency

### LINKED ISSUE #10096: [QA] Extend CLI test coverage for backup consistency
## Description
- [x] Add test coverage for bug https://github.com/owncloud/ocis/issues/9498.
  - Upload a file
  - Upload the same file again (using "replace" in the webui)
  - Upload the same file a third time
  - check consistency with ocis backup consistency
    -> There shouldn't be orphaned blob in the blobstore
    -> The file has only version
  >Note that this doesn't happen if you upload a file with different content but the same name. Versions are created as expected and there is no orphaned blobs.

  Test PR: https://github.com/owncloud/ocis/pull/10314

### PR DESCRIPTION
**Pull request by ghost**
## Description

This PR adds test coverage for issue https://github.com/owncloud/ocis/issues/9498

## Related Issue

- Fixes: https://github.com/owncloud/ocis/issues/10096

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/cliCommands/backupConsistency.feature [hunk 1]
```diff
@@ -23,4 +27,14 @@ Feature: backup consistency
     And the administrator has stopped the server
     When the administrator checks the backup consistency using the CLI
     Then the command should be successful
+    And the command output should contain "💚 No inconsistency found. The backup in '%storage_path%' seems to be valid."
+
+  @issue-9498
+  Scenario: check backup consistency after uploading a file multiple times
```
Comment (anon-pradip) [inline tests/acceptance/features/cliCommands/backupConsistency.feature:33]:
```suggestion
  Scenario: check backup consistency via CLI command after uploading a file multiple times
```

File: tests/acceptance/features/cliCommands/backupConsistency.feature [hunk 2]
```diff
@@ -4,11 +4,15 @@ Feature: backup consistency
   I want to check my data for inconsistencies
   So that I can backup my data
 
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
```
Comment (prashant-gurung899) [inline tests/acceptance/features/cliCommands/backupConsistency.feature:10]: As we are creating only one user(even though it's same):

```suggestion
     Given user "Alice" has been created with default attributes and without skeleton files
```

File: tests/acceptance/features/cliCommands/backupConsistency.feature [hunk 3]
```diff
@@ -23,4 +27,14 @@ Feature: backup consistency
     And the administrator has stopped the server
     When the administrator checks the backup consistency using the CLI
     Then the command should be successful
+    And the command output should contain "💚 No inconsistency found. The backup in '%storage_path%' seems to be valid."
+
+  @issue-9498
+  Scenario: check backup consistency after uploading a file multiple times via CLI command
```
Comment (saw-jan) [inline tests/acceptance/features/cliCommands/backupConsistency.feature:33]:
```suggestion
  Scenario: check backup consistency after uploading a file multiple times
```

File: tests/acceptance/features/cliCommands/backupConsistency.feature [hunk 4]
```diff
@@ -23,4 +27,14 @@ Feature: backup consistency
...
+    Given user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And the administrator has stopped the server
+    When the administrator checks the backup consistency using the CLI
+    Then the command should be successful
     And the command output should contain "💚 No inconsistency found. The backup in '%storage_path%' seems to be valid."
```
Comment (saw-jan) [inline tests/acceptance/features/cliCommands/backupConsistency.feature:40]: check the file versions as well. the bug report also mentions about it

File: tests/acceptance/features/cliCommands/backupConsistency.feature [hunk 5]
```diff
@@ -23,4 +25,18 @@ Feature: backup consistency
...
     Then the command should be successful
-    And the command output should contain "💚 No inconsistency found. The backup in '%storage_path%' seems to be valid."
\ No newline at end of file
+    And the command output should contain "💚 No inconsistency found. The backup in '%storage_path%' seems to be valid."
+
+  @issue-9498
+  Scenario: check backup consistency after uploading a file multiple times
+    Given user "Alice" has uploaded file with content "hello world" to "textfile.txt"
+    And user "Alice" has uploaded file with content "hello world" to "textfile.txt"
+    And user "Alice" has uploaded file with content "hello world" to "textfile.txt"
```
Comment (saw-jan) [inline tests/acceptance/features/cliCommands/backupConsistency.feature:34]: upload has to be done via TUS upload

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/cliCommands/backupConsistency.feature (hunk 5 above)
Comment (saw-jan) [reply tests/acceptance/features/cliCommands/backupConsistency.feature:34]: bug is not reproducible with normal upload

Comment (saw-jan) [reply tests/acceptance/features/cliCommands/backupConsistency.feature:34]: ahh, but it didn't fail when I did curl but it failed with web-upload

File: tests/acceptance/features/cliCommands/backupConsistency.feature (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/cliCommands/backupConsistency.feature:31]: `via CLI command` in a scenario title is rather unnecessary in this feature context (CLI feature context)

**Review by saw-jan — DISMISSED** [review]
:+1:

**Review by saw-jan — APPROVED** [review]

**Pull request merged**